executor/index_lookup_merge_join_test.go

```diff
@@ -64,3 +64,28 @@ func (s *testSuite9) TestIssue18631(c *C) {
 		"2 2 2 2 2 2 2 2",
 		"1 1 1 1 1 1 1 1"))
 }
+
+func (s *testSuiteWithData) TestIndexJoinOnSinglePartitionTable(c *C) {
+	// For issue 19145
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("set @try_old_partition_implementation = 1")
```

Comment:
This only fix the old implementation, it's enough for the 4.0 branch (This is a release blocker).
I'll file another PR to fix for the new implementation.